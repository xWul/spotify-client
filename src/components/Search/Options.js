/* LIBS */
import React from 'react'
import styled from 'styled-components'
/* CONFIG */
import { screens } from '../../theme/globalStyle'
/* COMPONENTS */
import Select from '../ui/Select'

const Content = styled.div`
  width: 40%;
  @media(max-width: ${screens.small}) {
    width: 100%
  }
`

const Options = (props) => {
  return (
    <Content>
      <Select
        {...props}
        value={props.selected}>
        <option value={'artist'}>Artists</option>
        <option value={'album'}>Albums</option>
        <option value={'track'}>Tracks</option>
      </Select>
    </Content>
  )
}

Options.defaultProps = {
  selected: 'artist'
}

export default Options
