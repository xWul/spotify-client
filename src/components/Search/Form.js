/* LIBS */
import React, { Component } from 'react'
import styled from 'styled-components'
/* CONFIG */
import { screens } from '../../theme/globalStyle'
/* COMPONENTS */
import Options from './Options'
import InputText from '../ui/InputText'
import Button from '../ui/Button'

const FormStyled = styled.form`
  display: flex;
  flex-direction: row
  width: 100%;
  @media(max-width: ${screens.small}) {
    flex-wrap: wrap;
  }
`

class Form extends Component {
  static defaultProps = {
    onSubmit: () => {},
    option: 'artist',
    query: ''
  }

  state = {
    option: '',
    query: ''
  }

  componentDidMount () {
    const { option, query } = this.props
    this.setState({option: option, query: query})
  }

  handleChange = ({target: {name, value}}) =>{
    this.setState({[name]: value})
  }

  submit = (event) => {
    event.preventDefault()
    const { query, option } = this.state
    this.props.onSubmit({ query, option })
  }

  render() {
    const { query, option } = this.state
    return (
      <FormStyled 
        method={'POST'}
        onSubmit={(event) => this.submit(event)}>       
        <Options
          onChange={this.handleChange}
          name={'option'}
          selected={option}/>
        <InputText
          onChange={this.handleChange}
          name={'query'}
          value={query}
          placeholder={'Search for Artists, Albums or Tracks...'} />
        <Button type={'submit'}>
          Search
        </Button>
      </FormStyled>
    )
  }
}

export default Form
