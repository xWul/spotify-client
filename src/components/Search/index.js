/* LIBS */
import React, { Component } from 'react'
import styled from 'styled-components'
/* CONFIG */
import { screens } from '../../theme/globalStyle'
/* COMPONENTS */
import Options from './Options'
import InputText from '../ui/InputText'
import Button from '../ui/Button'

const Form = styled.form`
  display: flex;
  flex-direction: row
  width: 100%;
  @media(max-width: ${screens.small}) {
    flex-wrap: wrap;
  }
`

class Search extends Component {
  static defaultProps = {
    onSubmit: () => {}
  }

  state = {
    option: 'artist',
    query: ''
  }

  handleChange = ({target: {name, value}}) =>{
    this.setState({[name]: value})
  }

  submit = (event) => {
    event.preventDefault()
    const { query, option } = this.state
    this.props.onSubmit({ query, option })
  }

  render() {
    const { query, option } = this.state
    return (
      <Form method={'POST'} onSubmit={(event) => this.submit(event)}>
        <Options
          onChange={this.handleChange}
          name={'option'}
          selected={option}/>
        <InputText
          onChange={this.handleChange}
          name={'query'}
          value={query}
          placeholder={'Search for Artists, Albums or Tracks...'} />
        <Button type={'submit'}>
          Search
        </Button>
      </Form>
    )
  }
}

export default Search
