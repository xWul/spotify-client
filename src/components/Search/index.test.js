import React from 'react'
import renderer from 'react-test-renderer'
import enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Search from './index'
import Options from './Options'

enzyme.configure({ adapter: new Adapter() })

describe('Search', () => {
  test('Search renderer without crash', () => {
    const component = renderer.create(
      <Search />,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Search changes state', () => {
    const handleChange = jest.fn()
    const event = { target: { name: 'option', value: 'test' } }
    const wrapper = shallow(<Search />)
  
    wrapper.find(Options).simulate('change', event)
    const state = wrapper.instance().state
    expect(state).toEqual({option: 'test', query: ""})
  })

  test('Search submit form', () => {
    const onSubmit = jest.fn()
    const component = mount(<Search  onSubmit={onSubmit} /> )
    component.simulate('submit')
    expect(onSubmit).toHaveBeenCalledTimes(1)
  })

})


