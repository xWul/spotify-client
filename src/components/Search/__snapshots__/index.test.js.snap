// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search renderer without crash 1`] = `
<form
  className="sc-gzVnrw cAeJnK"
  method="POST"
  onSubmit={[Function]}
>
  <div
    className="sc-htpNat farngo"
  >
    <div
      className="sc-bdVaJa LDKxg"
    >
      <select
        className="sc-bwzfXH fGrHko"
        name="option"
        onChange={[Function]}
        selected="artist"
        value="artist"
      >
        <option
          value="artist"
        >
          Artists
        </option>
        <option
          value="album"
        >
          Albums
        </option>
        <option
          value="track"
        >
          Tracks
        </option>
      </select>
    </div>
  </div>
  <div
    className="sc-bxivhb kBJQUN"
  >
    <input
      className="sc-ifAKCX hfbosk"
      name="query"
      onChange={[Function]}
      placeholder="Search for Artists, Albums or Tracks..."
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-EHOje kVUZme"
  >
    <button
      className="sc-bZQynM EzSAD"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
